Monika After Story/game/script-greetings.rpy

```diff
@@ -2594,15 +2595,13 @@ label greeting_upset:
             "Do you want something?",
         ]
 
-    $ upset_quip1 = renpy.random.choice(upset_greeting_quips_first)
+    $ upset_quip1 = renpy.substitute(renpy.random.choice(upset_greeting_quips_first))
 
-    # use renpy say so you can use text tags
-    show monika 2efc
-    $ renpy.say(m, upset_quip1)
+    m 2efc "[upset_quip1]"
 
     if renpy.random.randint(1,4) != 1:
-        $ upset_quip2 = renpy.random.choice(upset_greeting_quips_second)
-        $ renpy.say(m, upset_quip2)
+        $ upset_quip2 = renpy.substitute(renpy.random.choice(upset_greeting_quips_second))
+        m "[upset_quip2]"
```

Comment:
what was the point of changing all these lines? The comment said using `renpy.say` does substitute for you already.

Reply:
Because, as you can see in the next one, it's more efficient this way since we can use this instead and get rid of any 'show Monika' lines.

Reply:
i disagree on any sort of efficiency gain, since the `m exp "msg"` boils down to doing a `renpy.show` and `renpy.say` regardless. 
Using `renpy.say` abstracts the substitute line so you don't have to remember to format a choice prior to using text tags. Additionally, using "[this format]" prevents us from using formatting text tags (i.e: `{fast}`).
And typing `renpy.say` is much faster than `renpy.substitute`